src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -184,7 +184,8 @@ public String getDeterministHashFromFile(String chainTaskId) {
 
             if (deterministFilePath.toFile().exists()) {
                 byte[] content = Files.readAllBytes(deterministFilePath);
-                hash = bytesToString(Hash.sha256(content));
+                // if determinism.iexec file is already a byte32, no need to hash it again
+                hash = bytesToString(BytesUtils.isByte32(content) ? content : Hash.sha256(content));
```

Comment:
what if we had an arbitrary string of length 64 which is not necessarily a hash? do we have a problem?

Reply:
nope as long at it is deterministic :)